server/integration/Applications.js

```diff
@@ -55,19 +62,88 @@ exports.deleteApplication = async function deleteApplication(id) {
  * Accept the given application
  * @param id The user id of the application that is to be accepted.
  */
-exports.acceptApplication = async function acceptApplication(id) {
-  const validatedId = validateApp.validateId(id);
-  const applications = await db.loadUsersCollection();
-  await applications.updateOne({ _id: new ObjectId(validatedId) }, { $set: { status: "accepted" } }, { upsert: true });
+
+exports.acceptApplication = async function acceptApplication(id, timestamp) {
+  const session = await db.startSession();
+  const applications =  await db.loadUsersCollection();
+  await session.startTransaction();
+  const opts = { session, returnOriginal: false, new: true,
+                writeConcern: { w: "majority", wtimeout: 5000 } };
+  try{
+    const validatedId = validateApp.validateId(id);
+    const validatedTimestamp = validateApp.validateTimestamp(timestamp);
+    const commandResult = await applications.updateOne({ _id: new ObjectId(validatedId), timestamp: validatedTimestamp }, { $set: { status: "accepted", timestamp: timeStamp.generateTimestamp()}  }, opts)
+    if(commandResult.result.n == 0){
+      throw Error(Err.DatabaseErrors.UPDATE_UNSUCCESSFUL)
+    }
+
+    //await delayfunction.resolveAftermilliSeconds(8000);
+    await session.commitTransaction();
+    session.endSession();
+  }
+  catch(e){
+    console.log(e.message)
+    console.log(e.stack);
+    if(e.errorLabels){
+      console.log(e.errorLabels)
+    }
+    await session.abortTransaction();
+    session.endSession();
+    if(e.message === 'WriteConflict'){
+      throw Error(Err.DatabaseErrors.MONGO_WRITE_TRANSACTION_ERROR);
+    }
+    else{
+      throw e
```

Comment:
Really no reason to have such a short variable name. Would be better if it would be called "error", but it is not necessary.